.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,33 @@
+name: Run tests
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref_name }}
+  cancel-in-progress: true
+
+env:
+  GKE_CLUSTER: protocol-us-central1
+  GKE_ZONE: us-central1
+
+jobs:
+  go-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: "0"
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: "1.22.2"
+
+      - name: Run golangci-lint
```

Comment:
We need to install and configure it first. I don't remember exactly why, but we don't use the [official github action](https://golangci-lint.run/welcome/install/#github-actions) on our main repo.

If you don't want to look into this now — I'm happy to pick it up! In that case please remove `make go_lint` step.

Reply:
@okdas , I'd love to have a linter in this repo. Not sure how to proceed though.

Reply:
@red-0ne you can comment out this step, and I'll set up the `golangci-lint` next week then, would that work?